src/axom/core/Array.hpp

```diff
@@ -14,12 +14,45 @@
 #include "axom/core/IteratorBase.hpp"         // for Iterator
 
 // C/C++ includes
+#include <array>     // for std::array
 #include <iostream>  // for std::cerr and std::ostream
+#include <numeric>   // for std::inner_product
 
 namespace axom
 {
-// Forward declare the templated classes and operator function(s)
+// TODO: Add this as a non-type template parameter to Array/View
+enum MemoryResourceType
+{
+  Host,
+  Device,
+  Unified,
+  Pinned,
+  Constant,
+  File,
+  NoOp,
+  Shared,
+  Unknown
+};
```

Comment:
Minor: Should this go in `core/memory_management.hpp` ?

Reply:
This is just a placeholder for a later development, it should probably go in there eventually but I'll comment it out for now with a comment explaining what I intended to use it for